### ISSUE TITLE
How to align the test split .json files with the results in your ECCV paper (Table 4)?

### ISSUE THREAD

**Issue by ZhiZZhang** [aburns4/MoTIF#5]
I note that there are four test split files for MoTIF dataset in the folder `/motif_data_generation`: eccv_motif_app_seen_task_unseen_all.json, 
eccv_motif_app_seen_task_unseen_curr.json, 
eccv_motif_app_unseen_task_seen.json, 
eccv_motif_app_unseen_task_unseen.json.
Which one do you use for the results "App Seen" and "App Unseen" in Table 4 of your ECCV paper?

Besides, I also note your revised experiment results in your arXiv version after ECCV. But why the performance on "App Unseen" is higher that that on "App Seen"? (This has a different trend from the one in ECCV paper.)

Looking forwards to your reply. Thanks!

**Comment by aburns4** [comment]
Hi, yep there are four train/test splits. The difference between "app seen task unseen all" and "app seen task unseen curr" is whether the task was not seen in any app (all), or was seen in other apps, but not the current app (curr). The latter case is easier since models fine-tuned on MoTIF had the chance to see learn about the task and potential subtasks it requires, albeit in a different app environment.

App unseen task seen refers to the model never seeing any task in the app, so it is zero shot on that app. But the tasks have been seen before in other apps. Then, app unseen task unseen means neither were seen during fine-tuning, which is the most challenging setting.

Table 4 app unseen refers to "app unseen task unseen", and app seen refers to "app seen task unseen all". Table 14 in the supplementary has "app seen task unseen curr" and "app unseen task seen" to evaluate task generalization. 

Hope that clarifies things and feel free to ask follow up questions.

As for the results trends, the trend of app unseen performing **more poorly** than app seen (as expected), is consistent between the two versions. App unseen is ~1 point or less better only for the partial sequence accuracy of Seq2Act in our arXiv version which is the latest version of results. We hypothesize this could be from overfitting to the apps seen during training, or slight noise in our data.

**Comment by ZhiZZhang** [comment]
> For the results of Seq2Act on MoTIF, why are there APP seen? As claimed in your ECCV paper, the Seq2Act model is only trained on RicoSCA while being evaluated on MoTIF. So, how do you configure the evaluation of "App Seen" and "App Unseen" for Seq2Act model?

**Comment by aburns4** [comment]
I actually do not report App Seen / Task Unseen Curr or App Unseen / Task Seen for Seq2Act because it was not fine-tuned or trained jointly on MoTIF and would require this for tasks to have been seen in other apps than the test app, as you mentioned. The ability to study task generalization across app environments is a unique quality of MoTIF. Table 14 in the supplementary just reports Seq2Seq and MOCA models.

I will add the VLN methods to GitHub, it will just take me a bit more time.

As for the App Seen and App Unseen splits in Table 4, RicoSCA's apps are a superset of MoTIF's. So, all of the apps have been seen during training with the original RicoSCA data. For App Unseen, we removed the apps in RicoSCA training data that are in the MoTIF test splits.
